File: CS/MvcDashboardDataSources/Configuration/ObjectDataSourceConfigurator.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardWeb;

namespace MvcDashboardDataSources.Configuration {
    public class ObjectDataSourceConfigurator {
        public static void ConfigureDataSource(DashboardConfigurator configurator, DataSourceInMemoryStorage storage) {
            // Registers an Object data source.
            DashboardObjectDataSource objDataSource = new DashboardObjectDataSource("Object Data Source");
            objDataSource.DataId = "objectDataSource";
            storage.RegisterDataSource("objDataSource", objDataSource.SaveToXml());

            configurator.DataLoading += DataLoading;

        }
        private static void DataLoading(object sender, DataLoadingWebEventArgs e) {
            if (e.DataId == "objectDataSource") {
                e.Data = Invoices.CreateData();
            }
        }
    }
}

File: VB/MvcDashboardDataSources/Configuration/ObjectDataSourceConfigurator.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWeb

Namespace MvcDashboardDataSources.Configuration
	Public Class ObjectDataSourceConfigurator
		Public Shared Sub ConfigureDataSource(ByVal configurator As DashboardConfigurator, ByVal storage As DataSourceInMemoryStorage)
			' Registers an Object data source.
			Dim objDataSource As New DashboardObjectDataSource("Object Data Source")
			objDataSource.DataId = "objectDataSource"
			storage.RegisterDataSource("objDataSource", objDataSource.SaveToXml())

			AddHandler configurator.DataLoading, AddressOf DataLoading

		End Sub
		Private Shared Sub DataLoading(ByVal sender As Object, ByVal e As DataLoadingWebEventArgs)
			If e.DataId = "objectDataSource" Then
				e.Data = Invoices.CreateData()
			End If
		End Sub
	End Class
End Namespace